### PR TITLE
Computed appMenus option

### PR DESCRIPTION
**Pull request by samitiilikainen**
* Make `appMenus` optionally `IComputedValue` similarly as `trayMenus` already are (#6598)
* Allows extensions to dynamically show/hide menu items

### CODE REVIEW

**Review by aleksfront — APPROVED** [review]

**Review by panuhorsmalahti — APPROVED** [review]

File: src/features/application-menu/main/menu-registration.ts [hunk 1]
```diff
@@ -3,7 +3,9 @@
  * Licensed under MIT License. See LICENSE in root directory for more information.
  */
 import type { MenuItemConstructorOptions } from "electron";
+import type { IComputedValue } from "mobx";
 
-export interface MenuRegistration extends MenuItemConstructorOptions {
+export type MenuRegistration = {
```
Comment (Nokel81) [inline src/features/application-menu/main/menu-registration.ts:8]: Why is this required?

**Review by Nokel81 — COMMENTED** [review]
Sounds good. Though this sort of feature would be best used for when there is an unknown number of menu items instead of for hiding them.

File: src/features/application-menu/main/menu-registration.ts (hunk 1 above)
Comment (samitiilikainen) [reply src/features/application-menu/main/menu-registration.ts:8]: MenuItemConstructorOptions contains property visibility (boolean) and here we would like to add the option for it to be `IComputedValue<boolean>`

Comment (Nokel81) [reply src/features/application-menu/main/menu-registration.ts:8]: Sounds good, missed that on my phone

**Review by Nokel81 — APPROVED** [review]
